### PR TITLE
New main script for xcat autotest in jenkins environment

### PR DESCRIPTION
**Pull request by hu-weihua**
This pull request will replace pull request #884
1. relocate script to ~/xCAT-server/share/xcat/tools/jenkins/
2. add new features after review meeting

This is the main script for xcat autotest in jinkens environment.
This script is triggered by jinkens web page.
The using please refer to ``usage`` function, any comments are welcome

### CODE REVIEW

File: xCAT-server/share/xcat/tools/jenkins/xcatjktest [hunk 1]
```diff
@@ -0,0 +1,1355 @@
...
+        }
+    }
+    return 0;
+}
+
+#######################################
+# usage for arguments
+#######################################
+sub usage{
+    print "Usage:$pro_name - Run xcat test cases with jinkens.\n
```
Comment (whowutwut) [inline xCAT-server/share/xcat/tools/jenkins/xcatjktest]: jenkins spelled wrong

File: xCAT-server/share/xcat/tools/jenkins/xcatjktest [hunk 2]
```diff
@@ -0,0 +1,1355 @@
...
+  Explanation for the options:
+    -o: Requested, specify the test operating system and version\n
```
Comment (whowutwut) [inline xCAT-server/share/xcat/tools/jenkins/xcatjktest]: I think "Requested" should be changed to "Required"

File: xCAT-server/share/xcat/tools/jenkins/xcatjktest [hunk 3]
```diff
@@ -0,0 +1,1355 @@
...
+            }
+        }
+    }
+    return 0;
+}
+
+#######################################
+# usage for arguments
+#######################################
+sub usage{
```
Comment (whowutwut) [inline xCAT-server/share/xcat/tools/jenkins/xcatjktest]: General comment on the usage, can we only support long option names,  The short options does not really make sense with the letters...suggest:

```
--os
--cluster
--testcase
--project_name
--num
--xcat-core
--xcat-dep
--email
--quiet
```

Also do we support running the testcase using the daily builds? If we are specifying only a web address, I'm thinking we can only run weekly tests

Comment (tingtli) [inline xCAT-server/share/xcat/tools/jenkins/xcatjktest]: Now we use daily build on build server.  We can run daily test.
We will do refinement for the build part to support invoking build at the test time.

**Comment by whowutwut** [comment]
Thanks for the changes @hu-weihua  I'm ok for this to merge if @tingtli  is okay. 
I did not review any code function, just more externals.

**Comment by hu-weihua** [comment]
I open a new pull request #1062. please refer to the new one. close this old one.

**Pull request closed without merging**